{{cookiecutter.root_dir__update_if_you_intend_to_use_monorepo}}/{{cookiecutter.project_name}}/training/notebooks/TrainWithFeatureStore.py

```diff
@@ -30,11 +34,11 @@
 # MLflow experiment name.
 dbutils.widgets.text(
     "experiment_name",
-    "/{{cookiecutter.project_name}}/{{cookiecutter.experiment_base_name}}-test",
+    f"/{{cookiecutter.mlflow_experiment_parent_dir}}-{env}/{{cookiecutter.experiment_base_name}}-test",
```

Comment:
Actually, the `mlflow_experiment_parent_dir` is input from user. 
Now we are adding suffix to the parent dir name. It may be a bit unexpected. Can we move the `-{env}` to the experiment name instead of experiment folder?

Reply:
No because we generate the `mlflow_experiment_parent_dir` with a `-staging` suffix in the staging env, and a `-prod` suffix in the prod env to allow for single-workspace mode operation. So there is no directory with just `mlflow_experiment_parent_dir`

Reply:
@arpitjasa-db We may not need to update the parent dir with the {env}, since the experiment name is already different.
The mlflow-experiment.tf
```
resource "databricks_mlflow_experiment" "experiment" {
  name        = "${local.mlflow_experiment_parent_dir}/${local.env_prefix}{{cookiecutter.experiment_base_name}}"
  description = "MLflow Experiment used to track runs for {{cookiecutter.project_name}} project."
}
```
For example, if parent dir input from user is `parent-dir` and project_name is `project`. Then after the change you could have the following if all three points to the same workspace
 - /parent-dir-prod/prod-project : from terraform prod
 - /parent-dir-prod/project-test : manually run notebook and select prod
 - /parent-dir-staging/staging-project : from terraform staging or github test
 - /parent-dir-staging/project-test : manually run notebook and select prod
 - /dev-project : manually run notebook and select dev

I may have missed what the PR is trying to resolved. How about keeping the parent dir the same as user input?
- /parent-dir-prod/prod-project : from terraform prod
- /parent-dir-prod/staging-project  : from terraform staging
- /dev-project : manually run notebook, no matter what env is selected
- /test-project : github test

Reply:
If user manually run notebook `TrainWithFeatureStore.py` and select dev as env, I think it will throw error saying directory not found because dir `/parent-dir-dev/` doesn't exist.
As the list mentioned above, `/parent-dir-dev/project-test` is not there because `parent-dir-dev` will not be created by terraform.

Reply:
@mingyu89 in your second example you still call the directory `parent-dir-prod` but assuming you meant to keep it just `parent-dir`, the problem is in Terraform we create this directory via the modules. And in single-workspace mode we were creating two TF directory resources that pointed to the same directory which was overwriting each other's permissions and other resources. It's cleaner to just separate them. This was already done in a previous PR. 

In this PR, our flow is currently broken as I've seen during dogfooding, so I'm fixing it by passing in the env to the directory name, that's all this PR is for.

Reply:
> If user manually run notebook `TrainWithFeatureStore.py` and select dev as env, I think it will throw error saying directory not found because dir `/parent-dir-dev/` doesn't exist. As the list mentioned above, `/parent-dir-dev/project-test` is not there because `parent-dir-dev` will not be created by terraform.

Also for this yes, but we don't create a directory in the `dev` env anyways since we don't take in a dev workspace/create TF resources, so either way they'd need to manually create that directory in their dev env.

Reply:
@arpitjasa-db Ah I see it's so tricky. And in my example I meant `parent-dir`, updated.
If the parent dir must be different, how about keep the experiment so that we don't have `staging` and `prod` in two different places?

- /parent-dir-prod/project : from terraform prod
- /parent-dir-staging/project : from terraform staging
- /dev-project : manually run notebook, no matter what env is selected
- /test-project : github test

Reply:
Yeah the only issue with that would be in single-workspace mode, in the experiment observatory (or model registry if we do the same for models), the two experiments/models would look identical

Reply:
You are right. We only show the name on UI. Then my only `nit` is to but the "prod" on the same side, 
`/parent-dir-prod/project-prod` instead of `/parent-dir-prod/prod-project`.

Reply:
And we are using the staging workspace for `run-tests-fs.yml`, the parent dir makes sense.
Can we remove `dev` from this notebook? It has no use and only causes confusion.
After that we'll have
 - /parent-dir-prod/project-prod : from terraform prod
 - /parent-dir-prod/project-test : manually run notebook with "prod"
 - /parent-dir-staging/project-staging : from terraform staging
 - /parent-dir-staging/project-test : manually run notebook with "staging" or CI

Reply:
Yeah sure makes sense to me, I'll remove it and set default to `staging`